indigo_social/models.py

```diff
@@ -1,8 +1,61 @@
 # -*- coding: utf-8 -*-
+import os
+from io import BytesIO
+import sys
+
 from django.contrib.auth.models import User
+from django.core.files import File
+from django.core.files.uploadedfile import InMemoryUploadedFile
 from django.db import models
-from django.db.models.signals import post_save
+from django.db.models.signals import post_save, pre_save
 from django.dispatch import receiver
+from django.urls import reverse
+from PIL import Image
+
+from urllib.request import urlretrieve, urlcleanup
+
+from allauth.account.signals import user_signed_up
+from allauth.socialaccount.models import SocialLogin, SocialAccount
+from allauth.socialaccount.signals import pre_social_login, social_account_added, social_account_updated
+
+
+def user_profile_photo_path(instance, filename):
+    file_name, file_extension = os.path.splitext(filename)
+    return 'avatars/{}'.format(instance.id)
+
+
+def retrieve_social_profile_photo(user_profile, url):
+    """ Retrieve and save a retrieved user's profile photo from a url
+    """
+    try:
+        filename, _ = urlretrieve(url)
+        user_profile.profile_photo.save(filename, File(open(filename, 'rb')))
+    finally:
+        urlcleanup()
+
+
+def resize_photo(profile_photo, user):
```

Comment:
I suggest putting this into the UserProfile class, then it has all the context it needs.

Reply:
It can then resize itself, effectively.